### ISSUE TITLE
Parallel builds with virtual floppy fail in version 1.1.1 (working in 1.1.0)

### ISSUE THREAD

**Issue by taylor-madeak** [hashicorp/packer-plugin-vsphere#242]
#### Overview 

I am experiencing build "timed out waiting for IP address" failures caused by a "device not ready" error when attempting to access the A: drive (virtual floppy) on half of the jobs in a build when running parallel build jobs that don't leverage Packer parallelization.  

Issue may be related to issue #93, but I'm opening a separate bug report because this issue is specific to version 1.1.1.  Each process is creating and uploading a floppy image to the VM's unique folder, so I don't think overwriting is the issue here.  This appears to potentially be more of a race condition with each VM trying to access the same virtual floppy image.

**Last successful run:** 2022-12-02
**First failure:** 2022-12-06

I don't know if the guest OS is relevant to this, but I'm including it because this is the only base image we currently build where we use virtual floppies (because Windows won't source answer files from http) and parallelize to this degree.

Failures seem to have two modes:

- VMware Tools is never installed due to missing install script, which is provided by the floppy image and launched by AutoUnattend.
- Windows Setup never enters unattended mode due to missing AutoUnattend.xml.

In both cases Packer reports that the build times out waiting for an IP address.

#### Reproduction Steps

1. Run multiple builds as parallel jobs using different `vm_name` values for output.

### Packer Version

Using Docker image `hashicorp/packer:latest`.  Versions tested:

- 1.8.0
- 1.8.5

### Plugin Version and Builders

Plugin version 1.1.1.

- `vsphere-iso`

### VMware vSphere Version

vSphere version 7.0.3.00700

### Guest Operating System

These four are being built with parallel Bamboo jobs (order is important):

- Windows Server 2019 Datacenter  --> SUCCESSFUL
- Windows Server 2019 Datacenter Core --> FAILS
- Windows Server 2019 Standard --> SUCCESSFUL
- Windows Server 2019 Standard Core  --> FAILS

### Simplified Packer Buildfile

```
vcenter_server      = "REDACTED"
vcenter_cluster     = "REDACTED"
vcenter_datacenter  = "REDACTED"
vcenter_datastore   = "REDACTED"
vcenter_vmfolder    = "Templates/pipeline"
vm_network          = "REDACTED"
vm_os_type          = "windows2019srv_64Guest"
iso_paths           = [
    "[ISO] SW_DVD9_Win_Server_STD_CORE_2019_1809.1_64Bit_English_DC_STD_MLF_X22-02970.ISO",
    "[ISO] VMware-Tools-windows-12.1.0.20219665.iso"
]

source "vsphere-iso" "windows-server-standard" {
    # vCenter Config
    username                = var.vcenter_username
    password                = var.vcenter_password
    vcenter_server          = var.vcenter_server
    cluster                 = var.vcenter_cluster 
    datacenter              = var.vcenter_datacenter 
    datastore               = var.vcenter_datastore
    folder                  = var.vcenter_vmfolder
    
    # VM Hardware Config
    firmware                = "efi-secure"
    CPUs                    = 4
    RAM                     = 16384
    RAM_reserve_all         = false
    video_ram               = 131072
    usb_controller          = ["usb"]
    
    disk_controller_type    = ["pvscsi"]
    storage {
        disk_size             = 40960
        disk_thin_provisioned = true
    }

    floppy_files            = [
        var.windows_answer_file        
    ]
    floppy_dirs             = [
        "./scripts/bootstrap/"  # Contains VMware Tools install script
    ]

    network_adapters {
        network_card        = "vmxnet3"
        network             = var.vm_network
    }

    cdrom_type              = "sata"
    iso_paths               = var.iso_paths 

    # VM Options
    vm_name                 = var.vm_name  # Provided in CLI by the build job
    guest_os_type           = var.vm_os_type
    convert_to_template     = true
    remove_cdrom            = true
    create_snapshot         = false
    boot_wait               = "3s"
    boot_command            = ["<spacebar>"]
    shutdown_command        = "shutdown /s /t 10 /f /d p:4:1 /c \"Packer Shutdown\""
    shutdown_timeout        = "2h"
    
    # Communicator Config
    communicator            = "winrm"
    winrm_username          = var.winrm_username
    winrm_password          = var.winrm_password
    winrm_timeout           = "12h"
    winrm_insecure          = true
    winrm_use_ssl           = true
}

packer {
    required_plugins {
        # vSphere plugin
        vsphere = {
            version = "1.1.1"
            source = "github.com/hashicorp/vsphere"
        }
    }
}

build {
    sources = [
        "source.vsphere-iso.windows-server-standard"
    ]
}
```

### Operating System and Environment Details

Packer parallelizing is not being utilized here, instead we are building parallel with Atlassian Bamboo jobs.  This means I don't yet know if this affects a scenario that uses Packer parallelizing.

### Log Fragments and `crash.log` Files
Attached:

[packer_win_2019_dc-plugin_1_1_1-FAIL.log](https://github.com/hashicorp/packer-plugin-vsphere/files/10249949/packer_win_2019_dc-plugin_1_1_1-FAIL.log)

[packer_win_2019_dc_core-plugin_1_1_1-SUCCESS.log](https://github.com/hashicorp/packer-plugin-vsphere/files/10250240/packer_win_2019_dc_core-plugin_1_1_1-SUCCESS.log)

[packer_win_2019_dc_core-plugin_1_1_0-SUCCESS.log](https://github.com/hashicorp/packer-plugin-vsphere/files/10250286/packer_win_2019_dc_core-plugin_1_1_0-SUCCESS.log)

**Comment by tenthirtyam** [comment]
Do you see the same with `cd_content`?

**Comment by tenthirtyam** [comment]
Major changes were #178 and #240 that can be dug into if this can be reproduced - can other community members confirm the same?

**Comment by taylor-madeak** [comment]
> Do you see the same with `cd_content`?

Since these are Windows builds, `cd_content` isn't going to be a very useful test.  

> Does it always fail on Core?

Bamboo runs build jobs in parallel using separate build agents, but job order is at least somewhat a factor here: generally speaking, the first agent to kick off a build is the one that wins.  The other three jobs typically fail as illustrated below in this screenshot:

![image](https://user-images.githubusercontent.com/12554820/208493166-d8f72828-704a-4964-abbd-24ed4a475d83.png)

**Comment by tenthirtyam** [comment]
> > Do you see the same with `cd_content`?
> 
> Since these are Windows builds, `cd_content` isn't going to be a very useful test.

Thanks for the update - I personally use `cd_content` for Windows builds; hence my question.

**Comment by tenthirtyam** [comment]
Ah, I see you're not using `floppy_content`....

Thinking "outloud" can you specify the `floppy_img_path` does that resolve the issue - otherwise I think I used the same temporary name for the `.flp` file.

**Comment by taylor-madeak** [comment]
@tenthirtyam fair nuff, I'll check the Packer Docker image and see if the required binaries are present.  It'll be a while before I can set up to test this, though.  Any chance you were able to reproduce with `floppy_files`?

Can you provide an example for `floppy_img_path`?  It was my understanding that this would point to an existing `.flp` file in a datastore.  Are you suggesting that this can also control the name and path of the floppy image created by `floppy_*`?

**Comment by tenthirtyam** [comment]
> Can you provide an example for `floppy_img_path`? It was my understanding that this would point to an existing `.flp` file in a datastore. Are you suggesting that this can also control the name and path of the floppy image created by `floppy_*`?

Disregard.

**Comment by tenthirtyam** [comment]
> @tenthirtyam fair nuff, I'll check the Packer Docker image and see if the required binaries are present. It'll be a while before I can set up to test this, though. Any chance you were able to reproduce with `floppy_files`?

No, I've not had time to attempt reproducing this issue. I generally help the maintainers support the project in my spare time; limited whilst heading into the holidays.

**Comment by tenthirtyam** [comment]
> @tenthirtyam fair nuff, I'll check the Packer Docker image and see if the required binaries are present. It'll be a while before I can set up to test this, though. Any chance you were able to reproduce with `floppy_files`?

My guess at moment is that it's hitting a collision with the default `packer-tmp-created-floppy.flp` file name used in `StepAddFloppy`.

**Comment by taylor-madeak** [comment]
Oddly, I ran another test build this morning with the branch that I pinned the plugin to version 1.1.0 on and encountered the exact same failure mode (it's running right now).  Not sure if it's a name collision, but the paths appear unique to me:

```
packer-tmp-created-floppy.flp
1,440 KB
12/19/2022, 10:40:27 AM
Floppy Image
[dsc1-w1c3-vsan] 29bea063-bc50-2604-7a3d-b02628420370/packer-tmp-created-floppy.flp

packer-tmp-created-floppy.flp
1,440 KB
12/19/2022, 10:40:28 AM
Floppy Image
[dsc1-w1c3-vsan] 28bea063-accd-da08-9632-b0262841c820/packer-tmp-created-floppy.flp

packer-tmp-created-floppy.flp
1,440 KB
12/19/2022, 10:40:27 AM
Floppy Image
[dsc1-w1c3-vsan] 28bea063-9e52-d95e-cf00-b0262841f1f0/packer-tmp-created-floppy.flp
```

Could this be a vCenter issue?

**Comment by tenthirtyam** [comment]
Perhaps this is a duplicate of #93 if also seen on 1.1.0.

**Comment by tenthirtyam** [comment]
Duplicate of #93.

**Comment by tenthirtyam** [comment]
Duplicate of #93

**Comment by taylor-madeak** [comment]
I'm inclined to agree, I just reproduced the issue using Packer's built-in vSphere plugin.  I'm beginning to strongly suspect an issue on the vSphere side and not in the plugin itself.

**Comment by taylor-madeak** [comment]
Alright, I haven't nailed down the root cause yet, but I've narrowed it down to something environmental on our end.  Issue was not reproducible when running the build in our DEV cluster.  Closing this issue out.

**Comment by taylor-madeak** [comment]
Hey @tenthirtyam could this be caused by DRS migrating the VM to a different host (with the same VSAN) while Packer is waiting for an IP address?

<html><body>


12/19/2022, 1:32:40 PM | DRS migrated  BaseImage_WindowsServer2019DatacenterCore_hotfix/SDS-55262-7 from  dsc1-w1c3-vx03.anc.gci.com to dsc1-w1c3-vx04.anc.gci.com in cluster  dsc1-w1c3 in dsc1-w1-dc
-- | --



</body>
</html>

**Comment by tenthirtyam** [comment]
It's quite likely - I recall seeing similar GitHub Issues in the past for DRS.

Please note that you have identifying information in the comment above that you may not want public and you may wish do delete the comment.

**Comment by tenthirtyam** [comment]
Please note that it is still visible in the edited dropdown. Just want to make sure the comment is fully removed if sensitive.

**Comment by taylor-madeak** [comment]
I felt like none of it was especially sensitive, but I figure better safe than sorry. Thanks for looking out for me.
